internal/runtime/internal/controller/metrics.go

```diff
@@ -98,42 +98,82 @@ func (cm *controllerMetrics) Describe(ch chan<- *prometheus.Desc) {
 type controllerCollector struct {
 	l                      *Loader
 	runningComponentsTotal *prometheus.Desc
+	collectorHealth        *prometheus.Desc
 }
 
 func newControllerCollector(l *Loader, parent, id string) *controllerCollector {
 	return &controllerCollector{
 		l: l,
+		collectorHealth: prometheus.NewDesc(
+			"alloy_component_controller_health",
+			"Component health",
+			[]string{"health_type", "component_name"},
+			map[string]string{"controller_path": parent, "controller_id": id},
+		),
 		runningComponentsTotal: prometheus.NewDesc(
 			"alloy_component_controller_running_components",
-			"Total number of running components.",
-			[]string{"health_type"},
```

Comment:
I'm worried that some folks might rely on this label and this would be a breaking change for them.